packages/account/src/Sections/Verification/ProofOfAddress/proof-of-address-form.tsx

```diff
@@ -0,0 +1,415 @@
+                    });
+                    setIsLoading(false);
+                });
+            });
+        }, [account_settings, fetchResidenceList, fetchStatesList, is_eu]);
+
+        const changeable_fields = [...getChangeableFields()];
+
+        const validateFields = (values: TFormInitialValues) => {
+            (Object.entries(values) as ObjectEntries<TFormInitialValues>).forEach(
+                ([key, value]) => (values[key] = value.trim())
+            );
+
+            setFormState({ ...form_state, ...{ should_allow_submit: false } });
+            const errors: FormikErrors<TFormInitialValues> = {};
+            const validateValues = validate(errors, values);
+
+            const required_fields = ['address_line_1', 'address_city'];
+            validateValues(val => val, required_fields, localize('This field is required'));
+
+            const address_line_1_validation_result = validAddress(values.address_line_1, { is_required: true });
+            if (!address_line_1_validation_result.is_ok) {
+                errors.address_line_1 = address_line_1_validation_result.message;
+            }
+            const address_line_2_validation_result = validAddress(values.address_line_2);
+            if (!address_line_2_validation_result.is_ok) {
+                errors.address_line_2 = address_line_2_validation_result.message;
+            }
+
+            const validation_letter_symbol_message = localize(
+                'Only letters, space, hyphen, period, and apostrophe are allowed.'
+            );
+
+            if (values.address_city && !validLetterSymbol(values.address_city)) {
+                errors.address_city = validation_letter_symbol_message;
+            }
+
+            if (values.address_state && !validLetterSymbol(values.address_state) && states_list?.length < 1) {
+                errors.address_state = validation_letter_symbol_message;
+            }
+
+            if (values.address_postcode) {
+                if (!validLength(values.address_postcode, { min: 0, max: 20 })) {
+                    errors.address_postcode = localize(
+                        'Please enter a {{field_name}} under {{max_number}} characters.',
+                        {
+                            field_name: localize('Postal/ZIP code'),
+                            max_number: 20,
+                            interpolation: { escapeValue: false },
+                        }
+                    );
+                } else if (!validPostCode(values.address_postcode)) {
+                    errors.address_postcode = localize('Only letters, numbers, space, and hyphen are allowed.');
+                }
+            }
+
+            return errors;
+        };
+
+        const showForm = (should_show_form: boolean) => {
+            setFormState({ ...form_state, ...{ should_show_form } });
+        };
+
+        const isFieldsIssue = (errors: FormikErrors<TFormInitialValues>, values: TFormInitialValues) =>
```

Comment:
can we have a better name for this? something like hasFiledError

Reply:
it should highlight not only for errors, but also empty values.
I've got, will try to create proper name